trie/iterator.go

```diff
@@ -967,51 +964,131 @@ func (it *merkleTreeIterator) seek(path []byte) {
 			if it.err != nil {
 				return
 			}
-		} else {
-			break
 		}
+		break
 	}
 
-	if len(it.path) == 0 {
-		return // root node is not parent node
+	if _, ok := it.stack[len(it.stack)-1].(*merkleTreeIteratorParentNode); ok {
+		it.nextLeaf()
+	}
+
+	if leaf, ok := it.stack[len(it.stack)-1].(*merkleTreeIteratorLeafNode); ok {
+		switch bytes.Compare(path, zk.NewTreePathFromHashBig(common.BytesToHash(leaf.key))) {
+		case -1: // path < leaf path
+		loop1:
+			for it.prevLeaf() {
+				if it.err != nil {
+					return
+				}
+				switch bytes.Compare(path, zk.NewTreePathFromHashBig(common.BytesToHash(it.lastNodeAsLeaf().key))) {
+				case -1: // path < leaf path
+					continue
+				case 0: // path == leaf path
+					break loop1
+				case 1:
+					it.nextLeaf()
+					break loop1
+				}
+			}
+		case 0: // leaf path == path
+		case 1: // leaf path < path
+		loop2:
+			for it.nextLeaf() {
+				if it.err != nil {
+					return
+				}
+				switch bytes.Compare(path, zk.NewTreePathFromHashBig(common.BytesToHash(it.lastNodeAsLeaf().key))) {
+				case -1: // path < leaf path
+					it.prevLeaf()
```

Comment:
I think we need to check `it.err != nil` after call `it.prevLeaf()`
```suggestion
				switch bytes.Compare(path, zk.NewTreePathFromHashBig(common.BytesToHash(it.lastNodeAsLeaf().key))) {
				case -1: // path < leaf path
					it.prevLeaf()
					if it.err != nil {
						return
					}
```